src/defaultWhen.js

```diff
@@ -0,0 +1,25 @@
+import { curry } from 'ramda';
+
+
+/**
+ * Returns the second argument if predicate function returns `true`,
+ * otherwise the third argument is returned.
+ *
+ * @func defaultWhen
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.2.0|v2.2.0}
+ * @category Logic
+ * @sig  (a -> Boolean) -> b -> a -> a | b
+ * @param {!function} predicate The predicate function
+ * @param {*} defaultVal The default value
+ * @param {*} val `val` will be returned instead of `defaultVal` if predicate returns false
+ * @return {*} The `val` if predicate returns `false`, otherwise the default value
```

Comment:
```
* @param {b} defaultVal The default value
* @param {a} val `val` will be returned instead of `defaultVal` if predicate returns false
* @return {a|b} The `val` if predicate returns `false`, otherwise the default value

```

??

Reply:
I guess we've already talked about this in past. JSDOC deals with JavaScript types. It has no concept of type variables. I guess what we could do is to use `@typedef` to define them, but we can only define new types globally and not locally, which is a problem...

That is why we introduced `@sig` jsodc tag, to overcome this limitation and describe the function interface in Hindley-Millner signatures.

Reply:
Oh, sorry, got it